### PR TITLE
Use same thread to populate indexdb on mobile safari

### PR DESCRIPTION
**Pull request by justinbarry**
ref #3869

### CODE REVIEW

**Review by ghost — COMMENTED** [review]
This pull request has been automatically deployed to [FeaturePeek](https://featurepeek.com). 🥇

 This link will always point to this PR's latest changes: https://peek.run/o9phxqm

File: packages/augur-sdk/src/state/Sync.ts [hunk 1]
```diff
@@ -1,14 +1,16 @@
...
 // @TODO This function is currently a hack. Need to come up with a better solution
-export async function createAPIAndController(ethNodeUrl: string, account?: string, enableFlexSearch = false, pouchDBFactory:PouchDBFactoryType = PouchDBFactory({})) {
-  return create(ethNodeUrl, account, enableFlexSearch);
+export async function createAPIAndController(ethNodeUrl: string, account?: string, enableFlexSearch = false, databaseConfiguration:DatabaseConfiguration = {}) {
+  return create(ethNodeUrl, account, enableFlexSearch, databaseConfiguration);
 }
 
-export async function start(ethNodeUrl: string, account?: string, enableFlexSearch = false, pouchDBFactory:PouchDBFactoryType = PouchDBFactory({})): Promise<API> {
-  const { api, controller } = await create(ethNodeUrl, account, enableFlexSearch);
+export async function start(ethNodeUrl: string, account?: string, enableFlexSearch = false, databaseConfiguration:DatabaseConfiguration ={}): Promise<API> {
```
Comment (robert-e-davidson3) [inline packages/augur-sdk/src/state/Sync.ts:12]: formatting

```suggestion
export async function start(ethNodeUrl: string, account?: string, enableFlexSearch = false, databaseConfiguration: DatabaseConfiguration = {}): Promise<API> {
```

File: packages/augur-sdk/src/state/create-api.ts [hunk 2]
```diff
@@ -34,17 +35,17 @@ async function buildDeps(ethNodeUrl: string, account?: string, enableFlexSearch
   return { augur, blockAndLogStreamerListener, db };
 }
 
-export async function create(ethNodeUrl: string, account?: string, enableFlexSearch = false,  pouchDBFactory:PouchDBFactoryType = PouchDBFactory({})): Promise<{ api: API, controller: Controller }> {
-  const { augur, blockAndLogStreamerListener, db } = await buildDeps(ethNodeUrl, account, enableFlexSearch, pouchDBFactory);
+export async function create(ethNodeUrl: string, account?: string, enableFlexSearch = false,  databaseConfiguration:DatabaseConfiguration = {}): Promise<{ api: API, controller: Controller }> {
```
Comment (robert-e-davidson3) [inline packages/augur-sdk/src/state/create-api.ts:38]: formatting

```suggestion
export async function create(ethNodeUrl: string, account?: string, enableFlexSearch = false,  databaseConfiguration: DatabaseConfiguration = {}): Promise<{ api: API, controller: Controller }> {
```

**Review by robert-e-davidson3 — APPROVED** [review]
LGTM

Also, we should have a ticket for the additional work of replicating the in-memory database to an on-disk database, in the background.

**Review by pgebheim — APPROVED** [review]

**Comment by pgebheim** [comment]
@justinbarry is this ok to take the WIP off?

**Comment by justinbarry** [comment]
@pgebheim Done.

**Pull request merged**